# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoadContext.cpp
# Commit: c11cffaded9c
# Full Hash: c11cffaded9cfed53b04e5c7875278e9a9e55396
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r 0527d227c6b6 -r c11cffaded9c dom/script/ScriptLoadContext.cpp
--- a/dom/script/ScriptLoadContext.cpp	Fri Feb 18 12:45:52 2022 +0000
+++ b/dom/script/ScriptLoadContext.cpp	Fri Feb 18 12:45:53 2022 +0000
@@ -4,7 +4,6 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#include "ScriptLoadRequest.h"
 #include "GeckoProfiler.h"
 
 #include "mozilla/dom/Document.h"
@@ -15,15 +14,13 @@
 
 #include "js/OffThreadScriptCompilation.h"
 #include "js/SourceText.h"
+#include "js/loader/ScriptLoadRequest.h"
 
 #include "ModuleLoadRequest.h"
 #include "nsContentUtils.h"
 #include "nsICacheInfoChannel.h"
 #include "nsIClassOfService.h"
 #include "nsISupportsPriority.h"
-#include "ScriptSettings.h"
-
-using JS::SourceText;
 
 namespace mozilla {
 namespace dom {
@@ -134,7 +131,7 @@
   mOffThreadToken = nullptr;
 }
 
-void ScriptLoadContext::SetRequest(ScriptLoadRequest* aRequest) {
+void ScriptLoadContext::SetRequest(JS::loader::ScriptLoadRequest* aRequest) {
   MOZ_ASSERT(!mRequest);
   mRequest = aRequest;
 }
@@ -172,7 +169,8 @@
 
 bool ScriptLoadContext::IsPreload() const {
   if (mRequest->IsModuleRequest() && !mRequest->IsTopLevel()) {
-    ModuleLoadRequest* root = mRequest->AsModuleRequest()->GetRootModule();
+    JS::loader::ModuleLoadRequest* root =
+        mRequest->AsModuleRequest()->GetRootModule();
     return root->GetLoadContext()->IsPreload();
   }
 
@@ -182,7 +180,8 @@
 
 nsIGlobalObject* ScriptLoadContext::GetWebExtGlobal() const {
   if (mRequest->IsModuleRequest() && !mRequest->IsTopLevel()) {
-    ModuleLoadRequest* root = mRequest->AsModuleRequest()->GetRootModule();
+    JS::loader::ModuleLoadRequest* root =
+        mRequest->AsModuleRequest()->GetRootModule();
     return root->GetLoadContext()->GetWebExtGlobal();
   }
 
@@ -196,7 +195,8 @@
 
 nsIScriptElement* ScriptLoadContext::GetScriptElement() const {
   if (mRequest->IsModuleRequest() && !mRequest->IsTopLevel()) {
-    ModuleLoadRequest* root = mRequest->AsModuleRequest()->GetRootModule();
+    JS::loader::ModuleLoadRequest* root =
+        mRequest->AsModuleRequest()->GetRootModule();
     return root->GetLoadContext()->GetScriptElement();
   }
   nsCOMPtr<nsIScriptElement> scriptElement = do_QueryInterface(mElement);